vid/src/advz.rs

```diff
@@ -515,35 +496,25 @@ where
             .into(),
         );
 
-        // verify aggregate proof
+        // verify aggregate proofs
         //
         // some boilerplate needed to accommodate builds without `parallel`
         // feature.
-        let multiplicities = Vec::from_iter((0..multiplicity as usize));
+        let multiplicities = Vec::from_iter((0..multiplicity_usize));
         let polys_len = common.poly_commits.len();
         let multi_open_domain = self.multi_open_domain(multiplicity)?;
-        let verification_iter = parallelizable_slice_iter(&multiplicities).map(|i| {
-            let range = i * polys_len..(i + 1) * polys_len;
-            let aggregate_eval = polynomial_eval(
-                evals
-                    .get(range.clone())
-                    .ok_or_else(|| {
-                        VidError::Internal(anyhow::anyhow!(
-                            "share evals range {:?} out of bounds for length {}",
-                            range,
-                            evals.len()
-                        ))
-                    })?
-                    .iter()
-                    .map(FieldMultiplier),
-                pseudorandom_scalar,
-            );
+        let verification_iter = parallelizable_slice_iter(&multiplicities).map(|m| {
+            let evals_iter = evals.iter().skip(*m).step_by(multiplicity_usize);
+            let aggregate_eval =
+                polynomial_eval(evals_iter.map(FieldMultiplier), pseudorandom_scalar);
+            let domain_index = usize::try_from(share.index).map_err(vid)?
+                + m * usize::try_from(self.num_storage_nodes).map_err(vid)?;
```

Comment:
nit: we could use "convenience quantities" here as well.

Reply:
done in https://github.com/EspressoSystems/jellyfish/pull/678/commits/7a37893155ec163efa6a8e5553a3559609848c70